pycbc/filter/matchedfilter.py

```diff
@@ -765,20 +630,349 @@ def compute_u_val_for_sky_loc_stat(hplus, hcross, hphccorr,
+                                          corr_out=self.corr_cross_mem)
+
+        # The information on the complex side of this overlap is important
+        # we may want to use this in the future.
+        if not id(hplus) == self.cached_hplus_hcross_hplus:
+            self.cached_hplus_hcross_correlation = None
+        if not id(hcross) == self.cached_hplus_hcross_hcross:
+            self.cached_hplus_hcross_correlation = None
+        if not id(psd) == self.cached_hplus_hcross_psd:
+            self.cached_hplus_hcross_correlation = None
+        if self.cached_hplus_hcross_correlation is None:
+            hplus_cross_corr = overlap_cplx(hplus, hcross, psd=psd,
+                                           low_frequency_cutoff=self.flow,
+                                           high_frequency_cutoff=self.fhigh,
+                                           normalized=False)
+            hplus_cross_corr = numpy.real(hplus_cross_corr)
+            hplus_cross_corr = hplus_cross_corr / (hcross_norm*hplus_norm)**0.5
+            self.cached_hplus_hcross_correlation = hplus_cross_corr
+            self.cached_hplus_hcross_hplus = id(hplus)
+            self.cached_hplus_hcross_hcross = id(hcross)
+            self.cached_hplus_hcross_psd = id(psd)
+        else:
+            hplus_cross_corr = self.cached_hplus_hcross_correlation
+
+        snr = self._internal_snr_function(I_plus,I_cross,
+                                          hplus_cross_corr,
+                                          hpnorm=Iplus_norm,
+                                          hcnorm=Icross_norm,
+                                          out=self.snr_mem,
+                                          thresh=self.snr_threshold,
+                                          analyse_slice=stilde.analyze)
+        # FIXME: This should live further down
+        # Convert output to pycbc TimeSeries
+        delta_t = 1.0 / (self.tlen * stilde.delta_f)
+
+        snr = TimeSeries(snr, epoch=stilde.start_time, delta_t=delta_t,
+                         copy=False)
+
+        idx, snrv = events.threshold_real_numpy(snr[stilde.analyze],
+                                                self.snr_threshold)
+
+        if len(idx) == 0:
+            return [], 0, 0, [], [], [], [], 0, 0, 0
+        logging.info("%s points above threshold", str(len(idx)))
+
+
+        idx, snrv = events.cluster_reduce(idx, snrv, window)
+        logging.info("%s clustered points", str(len(idx)))
+        # erased self.
+        u_vals, coa_phase = self._internal_ext_params_function\
+            (I_plus.data, I_cross.data, hplus_cross_corr,
+             indices=idx+stilde.analyze.start, hpnorm=Iplus_norm,
+             hcnorm=Icross_norm)
+
+
+
+        return snr, Iplus_corr, Icross_corr, idx, snrv, u_vals, coa_phase,\
+                                      hplus_cross_corr, Iplus_norm, Icross_norm
+
+    def _internal_snr_function(self, hplus, hcross, hphccorr, **kwargs):
```

Comment:
Ok, I see why you are using these. Can you rename them to remove the redundant parts of the name, in particular 'internal' and 'function' don't seem necessary.

Reply:
Done.